src/rgw/rgw_process.cc

```diff
@@ -194,6 +273,7 @@ int process_request(rgw::sal::Store* const store,
 		    rgw::dmclock::Scheduler *scheduler,
                     string* user,
                     ceph::coarse_real_clock::duration* latency,
+                    std::shared_ptr<RateLimiter> ratelimit,
```

Comment:
why do you need a `shared_ptr`?
see: https://github.com/ceph/ceph/pull/42891/files#r767585678